narwhal/primary/src/proposer.rs

```diff
@@ -477,8 +477,11 @@ impl Proposer {
                             break;
                         }
 
+                        // Reset batch creation timer to now as we are attempting to resend an existing digest
+                        // that failed to be committed in a previous header.
+                        let created_at_timestamp = now();
                         digests_to_resend.extend(
-                            header.payload.iter().map(|(digest, worker)| (*digest, *worker, 0))
+                            header.payload.iter().map(|(digest, worker)| (*digest, *worker, created_at_timestamp))
```

Comment:
Should the time we set be `now` or the original time of the creation of the batch digest?

Reply:
Agree it would be better to reuse the original timestamp of the batch. We probably need some refactors to keep the information.